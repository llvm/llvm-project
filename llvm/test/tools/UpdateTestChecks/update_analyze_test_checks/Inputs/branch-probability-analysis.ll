; RUN: opt < %s -passes='print<branch-prob>' -disable-output 2>&1 | FileCheck %s

define void @test1() {
entry:
  br label %exit

exit:
  ret void
}
