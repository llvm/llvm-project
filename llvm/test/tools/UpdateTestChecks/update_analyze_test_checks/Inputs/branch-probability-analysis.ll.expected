; NOTE: Assertions have been autogenerated by utils/update_analyze_test_checks.py
; RUN: opt < %s -passes='print<branch-prob>' -disable-output 2>&1 | FileCheck %s

define void @test1() {
; CHECK-LABEL: 'test1'
; CHECK-NEXT:  ---- Branch Probabilities ----
; CHECK-NEXT:    edge entry -> exit probability is 0x80000000 / 0x80000000 = 100.00% [HOT edge]
;
entry:
  br label %exit

exit:
  ret void
}
